presto-spi/src/main/java/com/facebook/presto/spi/block/InterleavedBlockBuilder.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class InterleavedBlockBuilder
+        extends AbstractInterleavedBlock
+        implements BlockBuilder
+{
+    private final BlockBuilder[] blockBuilders;
+    private final InterleavedBlockEncoding blockEncoding;
+
+    private int positionCount;
+    private int currentBlockId;
```

Comment:
I would name this `currentBlock` or `currentBlockIndex`. `currentBlockId` sounds like it might be non-sequencial.